numpy/lib/format.py

```diff
@@ -265,6 +275,10 @@ def dtype_to_descr(dtype):
         replicate the input dtype.
 
     """
+    if _has_metadata(dtype):
+        warnings.warn("metadata on a dtype may be saved or ignored, but will "
+                      "raise if saved when read. Use another form of storage.",
```

Comment:
Do we have a tracking issue we can reference regarding making this work properly?

Reply:
I opened gh-15017

Reply:
Last question - do we want to reference that issue in the warning message?

Reply:
When we fix the issue the test will fail, which should bring us back to the warning. I think that is sufficient for this corner case.

Reply:
I meant more for pointing users to more information. Fine as is, just curious if putting issue numbers in warnings is a path we should look at going down.

Reply:
It is a good point, for now I think I will just copy the full warning text into the issue. Search engines will probably pick it up well.